### ISSUE TITLE
failed in resizing the volume or PV: unable to resize the volume: volume XXX does not exist

### ISSUE THREAD

**Issue by geoberle** [NetApp/trident#297]
We are running trident 18.10 with nas economy backend on top of OpenShift 3.11 and faced the following warning during a PVC resize

failed in resizing the volume or PV: unable to resize the volume: volume XXX does not exist

we verified with "tridentctl get volume XXX" that the volume exists but saw something interesting when getting details with -o yaml

```
items:
- Config:
    accessInformation:
      nfsPath: /_____/____
      nfsServerIp: _._._._
    accessMode: ReadWriteMany
    blockSize: ""
    cloneSourceSnapshot: ""
    cloneSourceVolume: ""
    cloneSourceVolumeInternal: ""
    encryption: ""
    fileSystem: ext4
    internalName: trident________
    name: ___
    protocol: file
    securityStyle: ""
    size: "537919488"
    spaceReserve: ""
    splitOnClone: ""
    storageClass: basic
    version: "1"
  backend: ontapnaseco__._._._
  orphaned: true
  pool: data01_01_SATA4000
```

the volume is orphaned. what does orphaned mean? how can we fix it? 

the qtree itself is there, can be mounted and works just fine.

**Comment by geoberle** [comment]
This bug was taken care of by the great NetApp support team and I'm happy to share the solution so others can benefit. 

Turns out the issue was self-made and not a bug in Trident after all. Turns out trident is handling the situation quite professionally with resilience in mind.

**TLDR**
Entirely my fault. Renaming the NetApp flexvolume to match the defined storage prefix fixed the issue.

**Background**
Within the backend definition you can define a storage prefix. Basically this is something you define if you want to point several trident backends to the same vserver - it keeps them separated. This storage prefix is also something trident uses to determine if it should feel responsible for a flexvolume and the containing qtrees or not (keep in mind we are talking about a NAS economy setup here). Our misstep was to redefine the storage prefix after we already created several PVs (and thus flexvolumes and qtrees). Therefore trident did not feel responsible for them anymore and marked them as "orphaned". The consequence was that trident was not managing the qtree anymore for  maintaining actions like "resize".

**How did we identity the issue?** 
Netapp support asked us to enable debug logging for a trident backend by specifying it like this...

`{
    "version": 1,
    "storageDriverName": "ontap-nas-economy",
     "debugTraceFlags": {"api": true}, ...
}`

By updating the backend with 
`tridntctl update backend <name> -f <backendfile>`
trident picks up the new config and show calls against the NetApp API in the logs. That enabled us to see qtree queries with a flexvolume prefix filter and explained why our "problematic qtree" at hand was marked as orphaned - it was simply not listed in the API results because of a wrongly named flexvolume.

**Solution**
Since only the internal qtree name is the identifying attribute, changing the flexvolume name is the way to go. Rename it in a way that it matches again your storage prefix. We kept the junction path intact so the PVC and consuming application at hand did not even notice the change. In order for trident to pickup the qtree again we simply updated the trident backend, which rendered the trident volume as "orphaned: false". From that point on, resizing worked like a charm.

Thanks to the patient NetApp support team. Their guidance did not only help to resolve this issue but did also elevate the understanding of tridents internal working within our team.
